### PR TITLE
Get USE_GPL_LIBS=0 working again.

### PR DESCRIPTION
**Pull request by ViralBShah**
Wrap all the FFTW docs in a USE_GPL_LIBS block.
JL_PRIVATE_LIBS also updated for USE_GPL_LIBS.

### CODE REVIEW

**Comment by Keno** [comment]
Wouldn't this be solved by just moving the docstrings into the appropriate file in base?

**Comment by ViralBShah** [comment]
Yes, but are we doing that right now? If so, that would solve this, although the Makefile patch is still required.

**Comment by JeffBezanson** [comment]
Yes, I think it's ok to start migrating doc strings into the source files where they belong.

**Comment by ViralBShah** [comment]
How about merge this to solve the immediate issue? Migration can be done soon afterwards as well.

**Comment by ViralBShah** [comment]
I really would prefer to merge this and then migrate, before other doc changes are checked in. Unless I hear an objection, I will do so in a few hours.

File: base/docs/helpdb.jl [hunk 1]
```diff
@@ -10692,7 +10768,7 @@ Store values from array `X` within some subset of `A` as specified by `inds`.
 
     setindex!(collection, value, key...)
 
-Store the given value at the given key or index within a collection. The syntax `a[i,j,...] = x` is converted by the compiler to `(setindex!(a, x, i, j, ...); x)`.
+Store the given value at the given key or index within a collection. The syntax `a[i,j,...] = x` is converted by the compiler to `setindex!(a, x, i, j, ...)`.
```
Comment (tkelman) [inline base/docs/helpdb.jl]: this is undoing a change in the wrong direction

Comment (ViralBShah) [inline base/docs/helpdb.jl]: I wonder how that snuck in. Will fix.

**Comment by JeffBezanson** [comment]
This needs to be reinstated.

**Pull request merged**